packages/common/src/types/filter.ts

```diff
@@ -171,4 +180,71 @@ export const convertDashboardFiltersToFilters = (
     return filters;
 };
 
+const isDashboardTileTargetFilterOverride = (
+    filter: string | Record<string, DashboardFieldTarget>,
+): filter is Record<string, DashboardFieldTarget> => typeof filter === 'object';
+
+export const convertDashboardFiltersParamToDashboardFilters = (
+    dashboardFilters: DashboardFiltersFromSearchParam,
+): DashboardFilters =>
+    Object.entries(dashboardFilters).reduce(
+        (result, [key, value]) => ({
+            ...result,
+            [key]: value.map((f) => ({
+                ...f,
+                ...(f.tileTargets && {
+                    tileTargets: f.tileTargets.reduce<
+                        Record<string, DashboardFieldTarget>
+                    >(
+                        (tileTargetsResult, tileTarget) => ({
+                            ...tileTargetsResult,
+                            ...(isDashboardTileTargetFilterOverride(tileTarget)
+                                ? {
+                                      [Object.keys(tileTarget)[0]]: {
+                                          fieldId:
+                                              tileTarget[
+                                                  Object.keys(tileTarget)[0]
+                                              ].fieldId,
+                                          tableName:
+                                              tileTarget[
+                                                  Object.keys(tileTarget)[0]
+                                              ].tableName,
+                                      },
+                                  }
+                                : {
+                                      [tileTarget]: {
+                                          fieldId: f.target.fieldId,
+                                          tableName: f.target.tableName,
+                                      },
+                                  }),
+                        }),
+                        {},
+                    ),
+                }),
+            })),
+        }),
+        { dimensions: [], metrics: [] },
+    );
+
+export const convertDashboardFiltersToParam = (
+    dashboardFilters: DashboardFilters,
+): DashboardFiltersFromSearchParam =>
+    Object.entries(dashboardFilters).reduce(
+        (result, [key, value]) => ({
+            ...result,
+            [key]: value.map((f) => ({
+                ...f,
+                ...(f.tileTargets && {
+                    tileTargets: Object.entries(f.tileTargets).map(
+                        ([tileTargetKey, tileTargetValue]) =>
+                            tileTargetValue.fieldId === f.target.fieldId
```

Comment:
missign a check for `tileTargetValue.tableName === f.target.tableName`

Reply:
sure, thanks! Adding that now and changing the name of the function!